cmd/vclusterctl/log/baselog_logger.go

```diff
@@ -0,0 +1,67 @@
+package log
+
+import (
+	"github.com/go-logr/logr"
+	"github.com/loft-sh/log"
+)
+
+type baseloggerLogSink struct {
+	name      string
+	keyValues map[string]interface{}
+	logger    log.BaseLogger
+}
+
+var _ logr.LogSink = &baseloggerLogSink{}
+
+func (*baseloggerLogSink) Init(info logr.RuntimeInfo) {
+}
+
+func (baseloggerLogSink) Enabled(level int) bool {
+	return true
+}
+
+func (l baseloggerLogSink) Info(level int, msg string, kvs ...interface{}) {
+	l.logger.Infof("%s\t", msg)
+	for k, v := range l.keyValues {
+		l.logger.Infof("%s: %+v  ", k, v)
+	}
+	for i := 0; i < len(kvs); i += 2 {
+		l.logger.Infof("%s: %+v  ", kvs[i], kvs[i+1])
+	}
+}
```

Comment:
This needs to be a proper implementation of the logger interface.

In logr, a message is an error, info, or info with a verbosity level (debug messages).
Per convention in loft + Kubernetes, infos level messages will have a level of 0, while any debug message will be higher than 0.

Also, the key-value pairs should be part of the log message, not separate log messages.

I'll update the github.com/loft-sh/log library to provide a method to convert a logger to a logr interface.